benchmarks/utils/monitor.py

```diff
@@ -34,13 +34,23 @@ def parse_func_repr(representation: str) -> Tuple[str, tuple, dict]:
     func_name, args_repr = representation[:-1].split("(", 1)
     args = []
     kwargs = {}
-    for arg in args_repr.split(", "):
-        if "=" in arg:
-            keyword, argument = arg.split("=", 1)
-            if keyword.isidentifier():
-                kwargs.update({keyword: argument})
-                continue
-        args.append(arg)
+    arg_list = args_repr.split(", ")
+    # Extract the keyword arguments from the positional arguments
+    for index in range(len(arg_list)):
+        # Check that the "," is not inside a parenthesis
+        # (this would mean that it is part of a string)
+        # If it is, then we merge the current string with the next one
+        # Until we find a string that is not inside a parenthesis
+        if arg_list[index].count("(") != arg_list[index].count(")"):
+            arg_list[index + 1] = arg_list[index] + ", " + arg_list[index + 1]
```

Comment:
I'm having a hard time understanding the logic of this section, could you please provide an example that didn't work with the old implementation and works with this one?

Reply:
It was to handle parameters with tuple parameters (like ngram_range), but it's not useful with the new changes.